OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -317,6 +320,10 @@ public void Load(Actor self, Actor a)
 
 		public void Killed(Actor self, AttackInfo e)
 		{
+			if (Info.EjectOnDeath)
+				while (!IsEmpty(self) && CanUnload())
```

Comment:
Is there a reason why you don't check for `DistanceAboveTerrain` here already? Then you wouldn't even need to run `SpawnPassengers`, including all the trait lookups and whatnot, in that situation.